### PR TITLE
Enable OptaPlanner 9 in the platform

### PR DESCRIPTION
**Pull request by rsynek**
Enable OptaPlanner 9.37.0.Final in the platform.

### CODE REVIEW

File: .github/mvn-settings.xml [hunk 1]
```diff
@@ -3,6 +3,14 @@
...
       <id>google-mirror</id>
       <repositories>
+        <repository>
+          <snapshots>
+            <enabled>false</enabled>
+          </snapshots>
+          <id>jboss-nexus</id>
+          <name>JBoss Nexus</name>
+          <url>https://repository.jboss.org/nexus/content/groups/public/</url>
+        </repository>
```
Comment (radtriste) [inline .github/mvn-settings.xml:13]:
```suggestion
        <repository>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
          <id>jboss-nexus</id>
          <name>JBoss Nexus</name>
          <url>https://repository.jboss.org/nexus/content/groups/public/</url>
        </repository>
        <repository>
          <snapshots>
            <enabled>false</enabled>
          </snapshots>
          <id>jboss-staging-nexus</id>
          <name>JBoss Staging Nexus</name>
          <url>https://repository.jboss.org/nexus/content/groups/kogito-public/</url>
        </repository>
```

**Comment by rsynek** [comment]
@aloubyansky Could you please take a look at this PR?

I added all generated files, yet the check complains that I need to run ./mvnw -Dsync (which I did).

The JDK 17 Build fails due to Camel integration tests, but that happens even with a clean main branch without my changes.

**Comment by aloubyansky** [comment]
For `org.drools` and `org.kie` groupIds that is.

**Comment by rsynek** [comment]
@aloubyansky There are the Camel failures I see locally as well. Is it a known problem? In other words, are we good to go with our release?

**Comment by aloubyansky** [comment]
I've seen the OpenTelemetry  failures in other PRs, I am not sure about the others. I'll re-run the CI.
BTW, it looks like you are adding assertj to the BOM, we used to have it in the quarkus-bom and decided to exclude it and let users pick their preferred version. It's also not that rare when it's not binary compatible from one release to another and may break compiled platform tests.

**Review by triceo — APPROVED** [review]

**Comment by rsynek** [comment]
@gsmet is there any way of excluding it from the platform but keeping it in the original optaplanner-build-parent? This dependency is used in nearly every OptaPlanner module. As we are in the middle of our Final release, introducing an additional level of parent pom.xml in our structure is also something I would rather avoid.

**Comment by aloubyansky** [comment]
Yes @rsynek, see https://github.com/quarkusio/quarkus-platform/blob/main/pom.xml#L641-L647

**Comment by rsynek** [comment]
@aloubyansky or @gsmet can you please advice how to re-trigger the CI here? We have fresh artifacts in the staging repo.

**Comment by aloubyansky** [comment]
I just did. Otherwise, a push would do it.

File: .github/mvn-settings.xml [hunk 2]
```diff
@@ -3,6 +3,22 @@
...
+        <repository>
+          <snapshots>
+            <enabled>false</enabled>
+          </snapshots>
+          <id>jboss-staging-nexus</id>
+          <name>JBoss Staging Nexus</name>
+          <url>https://repository.jboss.org/nexus/content/groups/kogito-public/</url>
+        </repository>
```
Comment (aloubyansky) [inline .github/mvn-settings.xml:21]: This will need to be removed before merging @rsynek

Comment (rsynek) [reply .github/mvn-settings.xml:21]: Of course, I will remove it once our artifacts hit the Maven central.

**Comment by rsynek** [comment]
@aloubyansky Only the open telemetry tests are failing. Are we good to go with finishing the OptaPlanner release?

CC @mbiarnes

**Review by aloubyansky — CHANGES_REQUESTED** [review]
Waiting for the extra repos to be removed.

**Comment by aloubyansky** [comment]
Yes, @rsynek

**Comment by rsynek** [comment]
> Waiting for the extra repos to be removed.

Done.

Thanks for your help with this PR.

**Comment by aloubyansky** [comment]
@rsynek would you mind rebasing please? Thanks.

**Comment by rsynek** [comment]
@aloubyansky commits squashed, and the build passed except for the telemetry tests.

**Review by aloubyansky — APPROVED** [review]
Thanks @rsynek

**Pull request merged**